nemo_curator/modules/semantic_dedup/clusteringmodel.py

```diff
@@ -128,7 +128,15 @@ def __call__(self, embeddings_dataset: DocumentDataset):
             embeddings_df = embeddings_df.repartition(
                 partition_size=self.partition_size
             )
-            embeddings_df = embeddings_df.to_backend("pandas").persist()
+
+            try:
+                embeddings_df = embeddings_df.to_backend("pandas").persist()
+            except IndexError:
+                raise RuntimeError(
+                    "DocumentDataset contains empty partitions. "
+                    "Please remove empty partitions from the dataset before running semantic deduplication."
+                )
```

Comment:
I am currently using this to catch https://github.com/NVIDIA/NeMo-Curator/issues/341. Another way is to check the lengths of the partitions, but I think that may be too computationally expensive. Not sure if there is a better way.